src/utils/components/SSHSecretSection/utils/components/SelectSecret.tsx

```diff
@@ -0,0 +1,86 @@
+import * as React from 'react';
+
+import { SecretModel } from '@kubevirt-ui/kubevirt-api/console';
+import { IoK8sApiCoreV1Secret } from '@kubevirt-ui/kubevirt-api/kubernetes';
+import { useKubevirtTranslation } from '@kubevirt-utils/hooks/useKubevirtTranslation';
+import { validateSSHPublicKey } from '@kubevirt-utils/utils/utils';
+import { ResourceLink, WatchK8sResult } from '@openshift-console/dynamic-plugin-sdk';
+import { Alert, AlertVariant, Select, SelectOption, SelectVariant } from '@patternfly/react-core';
+
+import Loading from '../../../Loading/Loading';
+import { decodeSecret } from '../utils';
+
+type SelectSecretProps = {
+  secretsResourceData: WatchK8sResult<IoK8sApiCoreV1Secret[]>;
+  selectedSecretName: string;
+  onSelectSecret: (secretName: string) => void;
+  id?: string;
+};
+
+const SelectSecret: React.FC<SelectSecretProps> = ({
+  secretsResourceData,
+  selectedSecretName,
+  onSelectSecret,
+  id,
+}) => {
+  const { t } = useKubevirtTranslation();
+  const [isSecretSelectOpen, setSecretSelectOpen] = React.useState(false);
+
+  const [secrets, secretsLoaded, secretsError] = secretsResourceData;
+  const sshKeySecrets = secrets
+    ? secrets?.filter((secret) => validateSSHPublicKey(decodeSecret(secret)))
+    : [];
+
+  const filterSecrets = (_, value: string): React.ReactElement[] => {
+    let filteredSecrets = sshKeySecrets;
+
+    if (value) {
+      const regex = new RegExp(value, 'i');
+      filteredSecrets = sshKeySecrets.filter((secret) => regex.test(secret?.metadata?.name));
+    }
+
+    return filteredSecrets.map((secret) => (
+      <SelectOption key={secret?.metadata?.name} value={secret?.metadata?.name}>
+        <ResourceLink kind={SecretModel.kind} name={secret?.metadata?.name} linkTo={false} />
+      </SelectOption>
+    )) as React.ReactElement[];
+  };
+
+  const onSelect = (event, newValue) => {
+    onSelectSecret(newValue);
+    setSecretSelectOpen(false);
+  };
+
+  if (!secretsLoaded) return <Loading />;
+
+  if (secretsError)
+    return (
+      <Alert title={t('Error')} isInline variant={AlertVariant.danger}>
+        {secretsError?.message}
+      </Alert>
+    );
+
+  return (
+    <Select
+      menuAppendTo="parent"
+      isOpen={isSecretSelectOpen}
+      onToggle={() => setSecretSelectOpen(!isSecretSelectOpen)}
+      onSelect={onSelect}
+      variant={SelectVariant.single}
+      onFilter={filterSecrets}
+      hasInlineFilter
+      selections={selectedSecretName}
+      placeholderText={t('--- Select secret ---')}
+      maxHeight={400}
+      id={id || 'select-secret'}
+    >
+      {sshKeySecrets?.map((secret) => (
```

Comment:
Not using filteredSecrets here?

Reply:
@upalatucci Why do we need both the filter function and the map here?

Reply:
the unfiltered data should be as array of `children`, while the `onFilter` prop is getting a function that returns array of filtered data as ReactElement which eventually calculated as the filtered options of the Select.
bottom line, this should be ok

Reply:
@avivtur Ah, that makes sense. It was late when I responded and I forgot that this was a filterable select component. 🤦🏼‍♂️ Thanks for clearing it up!

Reply:
Yes guys, just a little bit confusing but this is how `Select` works

Reply:
When looking on docs i can see onTypeaheadInputChanged prop on Select will give the input search value.
Then you dont need the filterChanged prop and just render instead of sshKeySecrets?.map just the filtered version. onTypeaheadInputChanged should active the filter. it will make a much more reasonable flow.

Reply:
<img width="1255" alt="image" src="https://user-images.githubusercontent.com/14824964/224060543-33ca9f39-2e14-4a17-ae5f-a3def8837f82.png">

Reply:
and will allow us to sperate between logic and elements in a single function

Reply:
If I understand correctly, you're saying to update the values passed as the children of the `Select` component by filtering on the input parameter inside `onTypeaheadInputChanged`. 

From my testing, the typeahead functionality doesn't work here. I found that if I removed the `ResourceLink` child in the `SelectOption` components it worked as expected, but I haven't been able to figure out a workaround to make it work while keeping the `ResourceLink` component. If you have any ideas I'd be happy to try them.

Reply:
Cool, is make a working version :)